src/adapters/net.cpp

```diff
@@ -130,6 +131,13 @@ namespace net {
     if (!*m_socketfd) {
       throw network_error("Failed to open socket");
     }
+    
+    m_status.mac = file_util::contents(NET_PATH + m_interface + "/address");
```

Comment:
I just tried changing the MAC address and the address does not change in polybar (because the network object is never recreated). We have to set the MAC address in the `query` function.

The commands I used for changing the MAC address:

```bash
sudo ip link set dev wlp3s0 down
sudo ip link set dev wlp3s0 address XX:XX:XX:XX:XX:XX
sudo ip link set dev wlp3s0 up
```

Reply:
Ok, I've moved the update code back into the query function